be/src/storage/lake/tablet_manager.cpp

```diff
@@ -912,4 +912,51 @@ StatusOr<TabletAndRowsets> TabletManager::capture_tablet_and_rowsets(int64_t tab
 
     return std::make_tuple(std::move(tablet_ptr), std::move(rowsets));
 }
+
+StatusOr<int64_t> TabletManager::collect_tablet_storage_size(int64_t tablet_id, int64_t curr_version,
+                                                             int64_t vacuum_version, int64_t min_version) {
+    auto root_dir = tablet_root_location(tablet_id);
+    auto data_dir = join_path(root_dir, lake::kSegmentDirectoryName);
+    auto version = curr_version;
+    int64_t total_size = 0;
+    ASSIGN_OR_RETURN(auto fs, FileSystem::CreateSharedFromString(root_dir));
+    while (version >= vacuum_version && version >= min_version) {
```

Comment:
is it possible the same dat file can be referenced in different version? so the same file will be counted multiple times. This `total_size` will be just a best-effort estimated size

Reply:
> is it possible the same dat file can be referenced in different version? so the same file will be counted multiple times. This `total_size` will be just a best-effort estimated size

The segment files will only be counted when the version is the latest version, because the tablet meta of the latest version contains information about all the currently valid segment files. At other times, only the sizes of the files that have not been deleted among the compaction input and orphan files will be calculated.